expr/func_groupbytags.go

```diff
@@ -0,0 +1,150 @@
+package expr
+
+import (
+	"bytes"
+	"errors"
+	"sort"
+	"strings"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncGroupByTags struct {
+	in         GraphiteFunc
+	aggregator string
+	tags       []string
+}
+
+func NewGroupByTags() GraphiteFunc {
+	return &FuncGroupByTags{}
+}
+
+func (s *FuncGroupByTags) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgString{val: &s.aggregator},
+		ArgStrings{val: &s.tags},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncGroupByTags) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncGroupByTags) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(s.tags) == 0 {
+		return nil, errors.New("No tags specified")
```

Comment:
can't we do this via a validator?

Reply:
For the `ArgStrings` I run the validators on each element, not the end result. Which would make more sense?

Reply:
validators on each element is definitely how it's applied for the other multi-types as well, and it makes sense because a validator is defined to take an `*expr` which means we can use the same Validator type for any of the types in expr/types.go
```
type Validator func(e *expr) error
```
perhaps we can add an additional validator attribute that will validate the entire `[]strings` slice (once all elements have been populated), but then I don't think we can still use the `Validator` type since the argument wouldn't be an `*expr` anymore.

Reply:
So, the parser actually returns a 400 if there are no tags/the tags weren't strings. So, it's likely that this check is redundant from the parse level.